pkg/meta/meta.go

```diff
@@ -1571,8 +1653,21 @@ func (i *HLastJobIterator) GetLastJobs(num int, jobs []*model.Job) ([]*model.Job
 	jobs = jobs[:0]
 	iter := i.iter
 	for iter.Valid() && len(jobs) < num {
+		match, err := IsJobMatch(iter.Value(), i.schemaNames, i.tableNames)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+
+		if !match {
+			err := iter.Next()
+			if err != nil {
+				return nil, errors.Trace(err)
+			}
+			continue
+		}
+
 		job := &model.Job{}
-		err := job.Decode(iter.Value())
+		err = job.Decode(iter.Value())
```

Comment:
The filter only skip the `job.Decode`. Is this the bottleneck in original problem? Please run a simple benchmark, I'm not sure it's worthy considering we are going to add many complex codes.

Reply:
I only did a test on my local machine with few ddl jobs(around 100k). I'll test it with more tables on TCMS later.

Reply:
I think 100k is enough. You can paste the result in PR description. For different filtering how much it will improve the performance.

Reply:
Since I only used 8 goroutine to create tables concurrently, the number of records in `tidb_ddl_jobs` is quite small. So I mainly tested the performance of querying history ddl jobs. I'll paste it later.

Reply:
Please check the performance of `where state = "synced" and db_name = "xxx"`. If there's not a huge improvement, I prefer we delete these optimization based on ExtractSchemaAndTableNameFromJob

Reply:
In the above tests using 100k jobs , the state of all jobs are "synced".（Please ignore the following differences. Because I tried to create more jobs after the test and I terminated the program, leading to 4 cancelled jobs.)
```mysql
mysql> select count(*) from information_schema.ddl_jobs;
+----------+
| count(*) |
+----------+
|   164156 |
+----------+
1 row in set (12.29 sec)

mysql> select count(*) from information_schema.ddl_jobs where state = "synced";
+----------+
| count(*) |
+----------+
|   164152 |
+----------+
1 row in set (9.59 sec)
```

Reply:
I mean check the effectiveness of filtering DB name and table name on history job, like run `... where state = "synced" and db_name = "any_db"` on old code and new code.